File: DownloadingImage/app/src/main/java/com/example/downloadingimage/MainActivity.java
package com.example.downloadingimage;

import androidx.appcompat.app.AppCompatActivity;

import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.os.AsyncTask;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.ImageView;

import java.io.InputStream;
import java.net.HttpURLConnection;
import java.net.URL;



public class MainActivity extends AppCompatActivity {
    public static  final String TAG = "Act";
    ImageView imageView;

    public class ImageDownloader extends AsyncTask<String,Void, Bitmap>{

        @Override
        protected Bitmap doInBackground(String... urls) {
            try {
                Log.i(TAG,"Downloading...");
                URL url = new URL(urls[0]);
                HttpURLConnection connection = (HttpURLConnection) url.openConnection();
                connection.connect();
                InputStream in = connection.getInputStream();
                Bitmap myBitmap = BitmapFactory.decodeStream(in);
                Log.i(TAG,"Image Downloaded!");
                return myBitmap;
            } catch (Exception e){
                e.printStackTrace();
                return null;
            }
        }
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        imageView = (ImageView) findViewById(R.id.imageView2);

        Button downloadButton = (Button) findViewById(R.id.downloadButton);
        downloadButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Log.i(TAG,"Button Pressed!");
                ImageDownloader task = new ImageDownloader();
                Bitmap myImage;
                try{
                    myImage = task.execute("https://upload.wikimedia.org/wikipedia/en/a/aa/Bart_Simpson_200px.png").get();
                    imageView.setImageBitmap(myImage);
                } catch (Exception e){
                    e.printStackTrace();
                }

            }
        });
    }
}

File: ButtonCllickApp/app/src/main/java/com/example/buttoncllickapp/MainActivity.kt
package com.example.buttoncllickapp

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.text.method.ScrollingMovementMethod
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast

class MainActivity : AppCompatActivity() {

    private var button: Button? = null
    private var textView: TextView? = null
    private var numTimesClicked = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val editText:EditText? = findViewById<EditText>(R.id.editText)
        button = findViewById<Button>(R.id.button)
        textView = findViewById<TextView>(R.id.textView2)
        textView?.text = ""
        editText?.setText("")
        textView?.movementMethod = ScrollingMovementMethod()
        /*
        button?.setOnClickListener(object : View.OnClickListener{
            override fun onClick(v: View?) {
                numTimesClicked++
//                Toast.makeText(this@MainActivity,numTimesClicked.toString(),Toast.LENGTH_SHORT).show()
                textView?.append("\n Button Clicked $numTimesClicked times")
            }
        })
         */
        button?.setOnClickListener {
            numTimesClicked += 1
            val str:String = editText?.text.toString()
            val msg:String = "$numTimesClicked. $str\n"
            textView?.append(msg)
            editText?.setText("")
        }
    }
}